### DISCUSSION TITLE
Add Constraint to MFXListView

### DISCUSSION THREAD

**Discussion by mspiropali** [palexdev/MaterialFX#231] [Archived]
Hello,

I am making a small project using MaterialFX and from the documentation, some components have MFXValidator and some do not. (i.e MFXListView). 
Is this intentional or do you plan to add it in the future?

Sorry if my question is vague, I am new to MaterialFX.

**Comment by palexdev** [comment]
> Hello,
> 
> I am making a small project using MaterialFX and from the documentation, some components have MFXValidator and some do not. (i.e MFXListView). 
> Is this intentional or do you plan to add it in the future?
> 
> Sorry if my question is vague, I am new to MaterialFX.

It is intentional. You usually want to have validators/constraints on input controls, it's rather unusual for a list to have such capabilities  
  
What are you trying to achieve?

**Reply by mspiropali** [reply]
I see what you mean and you are right, validators/constraints should exist on input controls.

I am using the MFXStepper with a MFXListView which has some predefined values. 
The user should select one value and the move to next toggle. If not, a validation error should display.
I believe this should be done via the validator on the MFXStepperToggle, but I cannot make it work. Maybe my approach is wrong?

Your "wrapNodeForValidation" function is awesome and I was trying to use it for this reason.

**Reply by palexdev** [reply]
So you've already been checking the demo files?

The stepper toggles already have validation capabilities as you can see in the createSteps() method
You just have to get the validators and add your constraints to it

**Reply by mspiropali** [reply]
Yes I have been reading the source code and trying the demo files for myself.

Yes I think I add the constraints, but I also need to use the "wrapNodeForValidation" in order to display the error label and not move to next step.

adding constraint: step2.getValidator().constraint("Select at least one", databasesList.getSelectionModel().selectionProperty().emptyProperty());
where databasesList is a MFXListView. This however does not show any error when pressing next without selecting a list item.

***EDIT***
I had a mistake in the constraint, should be databasesList.getSelectionModel().selectionProperty().sizeProperty().greaterThan(0));
but still I cannot use the wrapNodeForValidation to display the error bellow or next the list, as it is displayed under the TextField
